tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php

```diff
@@ -863,6 +864,84 @@ private function skipOci8AndMysqli()
             $this->markTestSkipped('Mysqli driver dont support this feature.');
         }
     }
+
+    public function testIteratorCallsFetchTheExpectedNumberOfTimes()
+    {
+        $sql = 'SELECT CURTIME(6) AS time_started, test_int, test_string FROM fetch_table';
+        $originalStmt = $this->_conn->prepare($sql);
+
+        // Mock the connection
+        $connectionMock = $this->getMockBuilder('Doctrine\\DBAL\\Driver\\Connection')
+            ->setMethods(['prepare', 'query'])
+            ->getMockForAbstractClass();
+
+        $stmtSpy = new StatementSpy($originalStmt);
```

Comment:
Cheers, struggling to find time to look at this right now, but hopefully will be able to pick it back up either later today or tomorrow. Appreciate your help though 👍